cmd/release-controller/http.go

```diff
@@ -925,3 +1015,248 @@ var extendedRelTime = []humanize.RelTimeMagnitude{
 func relTime(a, b time.Time, albl, blbl string) string {
 	return humanize.CustomRelTime(a, b, albl, blbl, extendedRelTime)
 }
+
+func (c *Controller) getReleaseCandidateList(releaseStreams ...string) (map[string]*ReleaseCandidateList, error) {
+	releaseCandidates := make(map[string]*ReleaseCandidateList)
+	if len(releaseStreams) == 0 {
+		return releaseCandidates, nil
+	}
+
+	releaseStreamTagMap, ok := c.findReleaseByName(true, releaseStreams...)
+	if !ok || len(releaseStreamTagMap) == 0 {
+		return releaseCandidates, fmt.Errorf("Unable to find all release streams: %s", strings.Join(releaseStreams, ","))
+	}
+
+	// next release version name for each release stream
+	next := make(map[string]*semver.Version)
+	// creation time of nightly/image that latest stable release in release stream was promoted from
+	latestPromotedTime := make(map[string]int64)
+	type stableRef struct {
+		from string
+		name string
+		time int64
+	}
+	stableReleases := make([]stableRef, 0)
+	stable, err := c.stableReleases(false)
+	if err != nil {
+		return releaseCandidates, err
+	}
+	for _, r := range stable.Releases {
+		if r.Release.Config.As == releaseConfigModeStable {
+			for _, tag := range r.Release.Source.Spec.Tags {
+				if tag.Annotations[releaseAnnotationSource] == fmt.Sprintf("%s/%s", r.Release.Source.Namespace, r.Release.Source.Name) {
```

Comment:
In golang style, we use "early exit" rather than "deep nesting".

So places where you do

```
if a == b {
  .. stuff
}
```

in a loop you'd instead do 

```
for ... {
  if a != b {
    continue
  }

  ... do stuff
}
```